components/api_request_helper/api_request_helper.cc

```diff
@@ -119,31 +110,25 @@ APIRequestHelper::Ticket APIRequestHelper::Request(
     const base::flat_map<std::string, std::string>& headers,
     const APIRequestOptions& request_options,
     ResponseConversionCallback conversion_callback) {
-  std::unique_ptr<URLLoaderHandler> handler = CreateURLLoaderHandler(
-      method, url, payload, payload_content_type,
-      request_options.auto_retry_on_network_change,
-      request_options.enable_cache, true /* allow_http_error_result*/, headers);
-
-  if (request_options.timeout) {
-    handler->url_loader_->SetTimeoutDuration(request_options.timeout.value());
-  }
+  auto iter = CreateRequestURLLoaderHandler(
+      method, url, payload, payload_content_type, request_options, headers,
+      std::move(callback));
```

Comment:
Combine setup of the request between Request and RequestSSE, so we don't forget anything (like we did already with SetTimeoutDuration)